### PR TITLE
docs: warn about running multiple local Ray instances

### PR DESCRIPTION
**Pull request by davidxia**
See https://discuss.ray.io/t/error-in-ray-job-submit-on-local-machine-if-multiple-clusters-are-running-at-the-same-time/14723/7?u=davidxia

## Why are these changes needed?

Will prevent user confusion

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by davidxia** [comment]
cc @anyscalesam

**Review by kevin85421 — APPROVED** [review]

**Comment by davidxia** [comment]
@kevin85421 @jjyao is this OK to merge?

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -1306,6 +1306,8 @@ def init(
             instance.
             4. If the provided address is "local", start a new local Ray
             instance, even if there is already an existing local Ray instance.
+            Running multiple local Ray instances is not supported and may have
+            undefined behavior.
```
Comment (jjyao) [inline python/ray/_private/worker.py:1310]: Why is it not supposed? I think this is supported.

Comment (davidxia) [reply python/ray/_private/worker.py:1310]: Should this be "Running multiple local Ray instances is not recommended and may have
undefined behavior." instead?

Comment (jjyao) [reply python/ray/_private/worker.py:1310]: You can do it as long as you provide unique ports for each Ray instance.

Comment (davidxia) [reply python/ray/_private/worker.py:1310]: @jjyao

1. Do you have an example Python snippet showing how to do this with `ray.init()`?
2. I understand it's possible, but I'm asking if it's not recommended?

Comment (davidxia) [reply python/ray/_private/worker.py:1310]: @jjyao just bumping again on the above questions

**Pull request merged**